chainset.go

```diff
@@ -100,3 +104,61 @@ func (cs chainSet) Start(ctx context.Context, testName string, additionalGenesis
 
 	return eg.Wait()
 }
+
+// TrackBlocks initializes database tables and polls for transactions to be saved in the database.
+// This method is a nop if dbFile is blank.
+// Expected to be called after Start.
+func (cs chainSet) TrackBlocks(ctx context.Context, testName, dbFile, gitSha string) error {
+	if len(dbFile) == 0 {
+		// nop
+		return nil
+	}
+
+	db, err := blockdb.ConnectDB(ctx, dbFile)
+	if err != nil {
+		return fmt.Errorf("connect to sqlite database %s: %w", dbFile, err)
+	}
+
+	if err := blockdb.Migrate(db); err != nil {
+		return fmt.Errorf("migrate sqlite database %s; deleting file recommended: %w", dbFile, err)
+	}
+
+	if len(gitSha) == 0 {
+		gitSha = "unknown"
+	}
+
+	testCase, err := blockdb.CreateTestCase(ctx, db, testName, gitSha)
+	if err != nil {
+		_ = db.Close()
+		return fmt.Errorf("create test case in sqlite database: %w", err)
+	}
+
+	// TODO (nix - 6/1/22) Need logger instead of fmt.Fprint
+	var eg errgroup.Group
+	for c := range cs {
+		c := c
+		name := c.Config().Name
+		finder, ok := c.(blockdb.TxFinder)
+		if !ok {
+			fmt.Fprintf(os.Stderr, `Chain %s is not configured to save blocks; must implement "FindTxs(ctx context.Context, height uint64) ([][]byte, error)"`+"\n", name)
+			return nil
+		}
+		eg.Go(func() error {
+			chaindb, err := testCase.AddChain(ctx, name)
```

Comment:
My assumption here is the chain name can be arbitrary in the event of same chains like gaia <-> gaia. I confirmed in our #engineering channel that testing the same chains will be common. 

I know in Cosmos chain name is canonical in the Chain Registry. However, in this context, I'm assuming it can be arbitray. E.g. `gaia1`, `gaia2`. 

May I should instead use the chain label?